react/week3/todolist/src/App.js

```diff
@@ -0,0 +1,217 @@
+    const [inputTodo, setInputTodo] = useState({
+        description: "",
+        deadline: ""
+    });
+    function setNewInput(inputValue, inputName) {
+        if (inputName === 'date') {
+            setInputTodo((prev) => {
+                return ({ ...prev, deadline: inputValue.getFullYear() + "-" + ((inputValue.getMonth() < 10) ? "0" : "") + (inputValue.getMonth() + 1) + "-" + ((inputValue.getDate() < 10) ? "0" : "") + inputValue.getDate() })
+            })
+        } else if (inputName === 'description') {
+            const { name, value } = inputValue;
+            setInputTodo((prev) => {
+                return ({ ...prev, [name]: value })
+            })
+        }
+    }
+
+    const { addMethod } = props
+    return (
+        <div className="add-todo-container">
+            <h3>Add a todo</h3>
+            <div className="inputs-container">
+                <DescriptionInput changeMethod={setNewInput} value={inputTodo.description} />
+                <DeadlineInput changeMethod={setNewInput} value={inputTodo.deadline} />
+                <div className="add-button-container">
+                    <button onClick={() => {
+                        addMethod(inputTodo)
+                        setInputTodo({
+                            description: "",
+                            deadline: null
+                        })
+                    }} className="add-button">Add todo</button>
+                </div>
+            </div>
+        </div>
+
+    )
+}
+
+function TodoListContainer() {
+    const [todo, setTodo] = useState([]);
+    const [id, setId] = useState(4);
+
+    useEffect(() => {
+        fetch("https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw")
+            .then(data => data.json())
+            .then(apiTodo => setTodo(prev => apiTodo))
+    }, [])
+    function incrementId() {
+        setId(prev => prev + 1);
+    }
+
+    function addTodo(newTodo) {
+        let { description, deadline } = newTodo
+        if (description.length === 0) {
+            description = undefined;
+        }
+        if (deadline === "") {
+            alert("Set a valid due date")
+            return
+        }
+
+        incrementId();
+        setTodo(prev => {
+            return (
+                [...prev, {
+                    id: id,
+                    description: description,
+                    deadline: deadline
+                }]
+            )
+        }
+
+        );
+    }
+
+    function deleteTodo(itemID) {
+        setTodo((prev) => {
+            return prev.filter((item, index) => {
+                return item.id !== itemID
+            });
+        })
+    }
+    return (
+        <div className="list-container">
+            <h1>Todo List</h1>
+            <TodoForm addMethod={addTodo} />
+            <TodoList todoList={todo} deleteMethod={deleteTodo} />
+        </div>
+    )
+}
+
+function TodoBorder(props) {
+    return (
+        <div className="todo-border" >
```

Comment:
this should be li, since you want to put it in ul and div can't be a child element of ul
```suggestion
        <li className="todo-border" >
```